lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -165,7 +167,23 @@ try {
         Exit-Json $result
     }
 
-    $principal = New-ScheduledTaskPrincipal -UserId "$user" -LogonType ServiceAccount
+    # Handle RunAs/RunLevel options for the task
+
+    if (!$store_password) {
+        # Create a ScheduledTaskPrincipal for the task to run under
+        $principal = New-ScheduledTaskPrincipal -UserId $user -LogonType S4U -RunLevel $runlevel -Id Author
```

Comment:
Does the default Windows service accounts like SYSTEM, NETWORK SERVICE still work until the `S4U` logon context?